src/Modules/Server.cpp

```diff
@@ -810,6 +822,38 @@ bool Server::Init() {
 	Memory::UnProtect((void *)(insn_addr + 2), 1);
 	*(char *)(insn_addr + 2) = 0x5C;
 
+#ifdef _WIN32
+	Server::EntityByIndex = (decltype(Server::EntityByIndex))Memory::Scan(server->Name(), "55 8B EC 8B 4D ? 33 C0 85 C9 7E ? 8B 15 ? ? ? ? 39 42");
+#else
+	Server::EntityByIndex = (decltype(Server::EntityByIndex))Memory::Scan(server->Name(), "8B 54 24 ? 85 D2 7E ? A1 ? ? ? ? 8B 40 ? 85 C0 74 ? C1 E2 04");
+#endif
+	g_EntityByIndexHook.SetFunc(Server::EntityByIndex);
+
+	uintptr_t PlayerClientCommand;
+#ifdef _WIN32
+	if (sar.game->Is(SourceGame_Portal2)) {
+		PlayerClientCommand = Memory::Scan(server->Name(), "55 8B EC 83 EC 78 53 56 57 8B 7D ? 83 3F 00");
+	} else {
+		PlayerClientCommand = Memory::Scan(server->Name(), "55 8B EC 83 EC 50 53 56 57 8B 7D ? 83 3F 00");
+	}
+#else 
+	if (sar.game->Is(SourceGame_Portal2)) {
+		PlayerClientCommand = Memory::Scan(server->Name(), "55 89 E5 57 56 53 81 EC 9C 00 00 00 8B 7D ? 8B 5D ? 8B 07");
+	} else {
+		PlayerClientCommand = Memory::Scan(server->Name(), "55 89 E5 57 56 53 83 EC 7C 8B 5D ? 8B 03 85 C0");
+	}
+#endif
+	uintptr_t code_a = (uintptr_t)(PlayerClientCommand) + 458;
+	if (*(uint8_t *)code_a == 0x7F) {
+		Memory::UnProtect((void *)code_a, 1);
+		*(uint8_t *)code_a = 0x74;
+	}
+	uintptr_t code_b = code_a + 32;
+	if (*(uint8_t *)code_b == 0x7F) {
+		Memory::UnProtect((void *)code_b, 1);
+		*(uint8_t *)code_b = 0x74;
+	}
```

Comment:
Can you add a comment on what all of this code here does?
The first offset into the function is unlikely to be the same on Windows and Linux.

Reply:
This turns conditional jumps (i forget if its gt or lt) for entity index 32 and 33 in the `signify` client command into jz which will never jump afaik

this fixes the ping not working for props, other hook just fixes a potential crash exploit for orange

Reply:
have you *tried* it on linux?